react/callbacks.rst

```diff
@@ -0,0 +1,198 @@
+.. _callbacks-label:
+
+===============================
+Use Callbacks To Delete An Item
+===============================
+
+Add Delete Button
+=================
+
+To be able to manage our faq entries we start by adding a delete button to
+remove an item from the list. Add the delete button to the view and create
+an empty :file:`onDelete` handler which is called when the button is pressed.
+
+..  admonition:: Solution
+    :class: toggle
+
+    ::
+
+        import React, { Component } from "react";
+        import PropTypes from "prop-types";
+        import "./FaqItem.css";
+
+        class FaqItem extends Component {
+          static propTypes = {
+            question: PropTypes.string.isRequired,
+            answer: PropTypes.string.isRequired
+          };
+
+          constructor(props) {
+            super(props);
+            this.toggle = this.toggle.bind(this);
+            this.onDelete = this.onDelete.bind(this);
+            this.state = {
+              show: false
+            };
+          }
+
+          toggle() {
+            this.setState({
+              show: !this.state.show
+            });
+          }
+
+          onDelete() {}
+
+          render() {
+            return (
+              <li className="faq-item">
+                <h2 onClick={this.toggle} className="question">
+                  {this.props.question}
+                </h2>
+                {this.state.show && <p>{this.props.answer}</p>}
+                <button onClick={this.onDelete}>Delete</button>
+              </li>
+            );
+          }
+        }
+
+        export default FaqItem;
+
+Write The onDelete Handler
+==========================
+
+Now that we have our dummy handler ready we need to add functionality to the
+handler. Since the list of Faq items is managed by our :file:`App` component we
+can't directly remove the item. Rewrite the :file:`FaqItem` component so that
+a unique identifier of the Faq item and a callback to remove the Faq item can be
```

Comment:
FAQ

Reply:
"can not" is easier to understand for everyone than "can't"